File: Java/src/com/dbc/algorithm/Array/MaximumNumberOfBallsInABox.java
package com.dbc.algorithm.Array;

import java.util.HashMap;
import java.util.Map;

public class MaximumNumberOfBallsInABox {
    public static int countBalls(int lowLimit, int highLimit) {
        int result = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = lowLimit; i <= highLimit; i++) {
            int num = i, remain = 0;
            while (num > 0) {
                remain += num % 10;
                num /= 10;
            }
            map.put(remain, map.getOrDefault(remain, 0) + 1);
            result = Math.max(map.get(remain), result);
        }

        return result;
    }
}

File: core/tmp/Python/array/maximum_number_of_balls_in_a_box.py
def sum_num(num: int) -> int:
    result = 0
    while num > 0:
        result += num % 10
        num //= 10
    
    return result

def countBalls(lowLimit: int, highLimit: int) -> int:
    result = 0
    remain = {}
    for i in range(lowLimit, highLimit + 1):
        num = sum_num(i)
        remain[num] = remain[num] + 1 if num in remain else 1
        result = max(result, remain[num])
        
    return result

if __name__ == "__main__":
    print(countBalls(lowLimit = 19, highLimit = 28))